backtester/eventhandlers/exchange/slippage/slippage.go

```diff
@@ -0,0 +1,48 @@
+package slippage
+
+import (
+	"math/rand"
+
+	gctorder "github.com/thrasher-corp/gocryptotrader/exchanges/order"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/orderbook"
+)
+
+// EstimateSlippagePercentage takes in an int range of numbers
+// turns it into a percentage
+func EstimateSlippagePercentage(maximumSlippageRate, minimumSlippageRate float64) float64 {
+	if minimumSlippageRate < 1 || minimumSlippageRate > 100 {
+		return 1
+	}
+	if maximumSlippageRate < 1 || maximumSlippageRate > 100 {
+		return 1
+	}
+
+	// the language here is confusing. The maximum slippage rate is the lower bounds of the number,
+	// eg 80 means for every dollar, keep 80%
+	randSeed := int(minimumSlippageRate) - int(maximumSlippageRate)
+	if randSeed > 0 {
+		result := float64(rand.Intn(randSeed)) // nolint:gosec // basic number generation required, no need for crypto/rand
+		return (result + maximumSlippageRate) / 100
+	}
+	return 1
+}
+
+// CalculateSlippageByOrderbook will analyse a provided orderbook and return the result of attempting to
+// place the order on there
+func CalculateSlippageByOrderbook(ob *orderbook.Base, side gctorder.Side, price float64) float64 {
```

Comment:
Unless I've misunderstood, I think this function will need a rework. Slippage function should take in an amount or $$ to be spent and then try figure out how much slippage will hurt placing that order. In the 'example of slippage' section in the following link they take an arbitrary number (100 shares) and calculate what the slippage is: https://www.investopedia.com/terms/s/slippage.asp

Reply:
The function should return for example 100 dollars of slippage per 100 bitcoins (1 dollar slippage per btc). Or 10 dollars of slippage per $100,000 spent.

Reply:
I would like to work with you on this later

Reply:
I've ammended this. It takes in the total funds available for the order simulates against the orderbook and then returns the price and amount. Is this what you are after?

Reply:
Yeah